### PR TITLE
[#11553] update openapi

### LINKED ISSUE #11553: Update the dependencies for openAPI generation [2]
### Problem Description
While working on #10251, it became clear that we need to update the dependencies of the openAPI generation. This has been done as part of #11202, however, as this PR will not land soon, we need to extract the work done there to unblock #10251.

### Proposed Change

- [x] Extract all openAPI related changes from the POMs in #11202.
- [x] Make sure generation sill works.

### PR DESCRIPTION
**Pull request by JonasCir**
Fixes #11553

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

File: sormas-rest/README.md [hunk 1]
```diff
@@ -9,7 +9,9 @@ This is a one-stop-shop for all systems that need access to the SORMAS data:
 * Synchronization of data with other surveillance or to data analysis systems
 
 ## Authentication
-Access to the API is by default restricted by HTTP Basic authentication. Using OIDC/OAUTH2/Bearer authentication is also possible depending on how keycloak is setup. See , it can als use Bearer authentication. See [Authentication & Authorization](https://github.com/hzi-braunschweig/SORMAS-Project/wiki/Authentication-&-Authorization#keycloak). 
+Access to the API is by default restricted by HTTP Basic authentication. Using OIDC/OAUTH2/Bearer authentication is
+also possible depending on how keycloak is setup. See , it can als use Bearer authentication.
```
Comment (MartinWahnschaffe) [inline sormas-rest/README.md:13]: > See , it can als use Bearer authentication.

Not sure what this was about

File: sormas-rest/README.md [hunk 2]
```diff
@@ -39,4 +41,356 @@ The specification files are created at the path specified above.
 
 ---
 
-<a id="SwaggerAnnotations"></a>\[1] Swagger Annotations Guide on Github: <https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations>
+<a id="SwaggerAnnotations"></a>\[1] Swagger Annotations Guide on Github:
+<https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations>
+
+
+## External Visist API
```
Comment (MartinWahnschaffe) [inline sormas-rest/README.md:48]:
```suggestion
## External Visits API
```

File: sormas-rest/src/main/resources/openApiConfig.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+openAPI:
+  info:
+    title: SORMAS REST API
```
Comment (MartinWahnschaffe) [inline sormas-rest/src/main/resources/openApiConfig.yml:3]: Let's add the following based on https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Integration-and-Configuration#configuration-file

```suggestion
    title: SORMAS REST API
    description: https://github.com/hzi-braunschweig/SORMAS-Project/tree/development/sormas-rest
    license:
      name: GPL v3
      url: https://www.gnu.org/licenses/gpl-3.0.html
```


Also we should include a version (required field), but I'm not sure what's the best approach here. Would be nice to take the current version from sormas-base pom.

File: sormas-app/app/build.gradle [hunk 4]
```diff
@@ -7,6 +7,11 @@ ext.sormasVersion = getVersionName()
 ext.keystoreProperties = loadKeystoreProperties()
 
 android {
+    packagingOptions{
```
Comment (MartinWahnschaffe) [inline sormas-app/app/build.gradle:10]: Why did this have to be removed? We should probably add a comment

File: sormas-app/gradle.properties [hunk 5]
```diff
@@ -10,7 +10,7 @@
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
 # Default value: -Xmx10248m -XX:MaxPermSize=256m
-org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
```
Comment (MartinWahnschaffe) [inline sormas-app/gradle.properties:13]: Why did this have to be removed? Doesn't it mean that the default value of 256m is used?

**Review by MartinWahnschaffe — CHANGES_REQUESTED** [review]

**Comment by MartinWahnschaffe** [comment]
@JonasCir Please also remove the openapi folder with the external visit swagger files OR let's generate the current swagger files into that directory.

File: sormas-rest/README.md (hunk 1 above)
Comment (JonasCir) [reply sormas-rest/README.md:13]: removed

File: sormas-app/app/build.gradle (hunk 4 above)
Comment (JonasCir) [reply sormas-app/app/build.gradle:10]: fixed

File: sormas-app/gradle.properties (hunk 5 above)
Comment (JonasCir) [reply sormas-app/gradle.properties:13]: Java 8 VM no longer supports `MaxPermSize`. It doesn't work locally for me and I really don't get how this gets through the gradle CI as it uses openjdk 11...

Comment (JonasCir) [reply sormas-app/gradle.properties:13]: done

**Comment by JonasCir** [comment]
Update https://github.com/hzi-braunschweig/SORMAS-Project/blob/7e46f92ff1afc353cde9afa2c71bca9e8c70f5e7/.github/workflows/ci.yml#L112

File: sormas-rest/README.md (hunk 2 above)
Comment (JonasCir) [reply sormas-rest/README.md:48]: fixed

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Comment by JonasCir** [comment]
@MartinWahnschaffe You can take a look, I used a quite nice mechanism for the version number :)

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=11554

**Review by MartinWahnschaffe — APPROVED** [review]

**Pull request merged**